### PR TITLE
fix: service resources not updated when redeploying

### PR DESCRIPTION
**Pull request by shanewxy**
**Problem:**

When deploying services fails for the first time(e.g. Fail to create VPC), the instance of some resources are empty. Upgrading the service does not cause the instance field updated.

**Solution:**

When getting a new service revision, in addition to resource creation and deletion, update the instance Edge for the existing resources as well.

**Related Issue:**
#1283

### CODE REVIEW

**Review by rainfd — COMMENTED** [review]
Why is this operation not in the transaction?

File: pkg/apis/servicerevision/extension.go
Comment (rainfd) [inline pkg/apis/servicerevision/extension.go:1]: Why is this operation not in the transaction?

File: pkg/apis/servicerevision/extension.go [hunk 1]
```diff
@@ -163,7 +163,15 @@ func manageResources(
 	for _, c := range recordRess {
 		k := dao.ServiceResourceGetUniqueKey(c)
 		if observedRessIndex[k] != nil {
+			c.Edges.Instances = observedRessIndex[k].Edges.Instances
+
+			err = dao.ServiceResourceInstancesEdgeSave(ctx, modelClient, c)
```
Comment (rainfd) [inline pkg/apis/servicerevision/extension.go:168]: Why is this operation not in the transaction?

Comment (shanewxy) [reply pkg/apis/servicerevision/extension.go]: Updated

**Review by rainfd — APPROVED** [review]
LGTM

**Review by alexcodelf — APPROVED** [review]
LGTM

**Pull request merged**